packages/@aws-cdk/aws-eks/lib/k8s-cluster-autoscaler.ts

```diff
@@ -0,0 +1,389 @@
+   */
+  readonly nodeGroups: autoscaling.AutoScalingGroup[];
+
+  /**
+   * The version of the Cluster Autoscaler to deploy.
+   *
+   * @default - the defualt version is v1.14.6
+   */
+  readonly version?: string;
+
+}
+
+/**
+ * The Cluster Autoscaler Construct. This will create a new IAM Policy, add labels to the ASGs, and
+ * deploy the Cluster Autoscaler manifest.
+ */
+export class KubernetesClusterAutoscaler extends cdk.Construct implements IKubernetesClusterAutoscaler {
+
+  /**
+   *  The cluster this autoscaler is assosciated with
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  public readonly cluster: Cluster;
+
+  /**
+   *  The IAM policy created by this construct.
+   */
+  public readonly policy: iam.IPolicy;
+
+  /**
+   * The Kubernetes Resource that defines the Cluster Autoscaler K8s resources.
+   */
+  public readonly clusterAutoscaler: KubernetesResource;
+
+  /**
+   * Constructs a new instance of the Cluster Autoscaler.
+   *
+   * @param scope cdk.Construct
+   * @param id string
+   * @param props ClusterAutoscalerProps
+   */
+  constructor(scope: cdk.Construct, id: string, props: KubernetesClusterAutoscalerProps) {
+    super(scope, id);
+
+    this.cluster = props.cluster;
+
+    const clusterName = props.cluster.clusterName;
+    const version = props.version === undefined ? 'v1.14.6' : props.version;
+
+    // define the cluster autoscaler policy statements
+    // https://docs.aws.amazon.com/en_pv/eks/latest/userguide/cluster-autoscaler.html#ca-create-ngs
+    const policyStatement = new iam.PolicyStatement();
+    policyStatement.addResources('*');
+    policyStatement.addActions(
+      'autoscaling:DescribeAutoScalingGroups',
+      'autoscaling:DescribeAutoScalingInstances',
+      'autoscaling:DescribeLaunchConfigurations',
+      'autoscaling:DescribeTags',
+      'autoscaling:SetDesiredCapacity',
+      'autoscaling:TerminateInstanceInAutoScalingGroup',
+      'ec2:DescribeLaunchTemplateVersions'
+    );
+
+    // create the policy based on the statements
+    const policy = this.policy = new iam.Policy(this, 'cluster-autoscaler-policy', {
+      policyName: 'ClusterAutoscalerPolicy',
+      statements: [ policyStatement ]
+    });
+
+    // loop through all of the node groups and attach the policy
+    props.nodeGroups.forEach(element => {
+      cdk.Tag.add(element, 'k8s.io/cluster-autoscaler/' + clusterName, 'owned', { applyToLaunchedInstances: true });
+      cdk.Tag.add(element, 'k8s.io/cluster-autoscaler/enabled', 'true', { applyToLaunchedInstances: true });
+      policy.attachToRole(element.role);
+    });
+
+    // define the Kubernetes Cluster Autoscaler manifests
+    this.clusterAutoscaler = new KubernetesResource(this, 'k8s-cluster-autoscaler-manifest', {
+      cluster: props.cluster,
+      manifest: [
+        {
+          apiVersion: 'v1',
+          kind: 'ServiceAccount',
+          metadata: {
+            name: 'cluster-autoscaler',
+            namespace: 'kube-system',
```

Comment:
Extract `namespace` and `name` to variables. Don't repeat yourself (now that you can avoid it)

Reply:
Oh yeah :) Good call out.